test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/DotnetCommand.cs

```diff
@@ -15,6 +16,35 @@ public DotnetCommand()
         public DotnetCommand(string dotnetUnderTest)
             : base(dotnetUnderTest)
         {
+            // Some scripts (e.g. RunCsc/RunVbc) depend on the correct `dotnet`
+            // being in the PATH. When tests use `DotnetUnderTest.WithBackwardsCompatibleRuntimes`,
+            // this means that the up-to-date dotnet command is resolved by the
+            // old-runtime-targeting RunCsc/Roslyn, causing compatability issues.
+            // So, detect if we're running a dotnet that's not just a plain "dotnet",
+            // and if so, make sure it's on the PATH.
```

Comment:
Agreed. I recall talking this over with @agocke, so they can comment more, but - this is a result of a major fundamental issue that we need to figure out of "how do we pass the host down to client code". That requires fairly significant design work, and as of right now I've left it as this cludge. (Setting a special variable goes against the purpose of decoupling the CLI and Roslyn more)